From 32f7f7dfd8a1b1afdb93703b5ceb9c4605b20b21 Mon Sep 17 00:00:00 2001
From: Cory Fields <cory-nospam-@coryfields.com>
Date: Fri, 7 Apr 2023 18:01:31 +0000
Subject: [PATCH 1/2] bootstrap: don't over-quote compiler variables

---
 bootstrap | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/bootstrap b/bootstrap
index 416a3d6780..ee8769f3ec 100755
--- a/bootstrap
+++ b/bootstrap
@@ -744,7 +744,7 @@ cmake_try_run ()
   echo "----------  file   -----------------------"
   cat "${TESTFILE}"
   echo "------------------------------------------"
-  "${COMPILER}" ${FLAGS} "${TESTFILE}" -o "${TMPFILE}"
+  ${COMPILER} ${FLAGS} "${TESTFILE}" -o "${TMPFILE}"
   RES=$?
   if [ "${RES}" -ne "0" ]; then
     echo "Test failed to compile"
@@ -1156,7 +1156,7 @@ mkdir "${cmake_bootstrap_dir}/${TMPFILE}"
 cd "${cmake_bootstrap_dir}/${TMPFILE}"
 echo '
 test: test.c
-	"'"${cmake_c_compiler}"'" '"${cmake_ld_flags} ${cmake_c_flags}"' -o test test.c
+	'"${cmake_c_compiler}"' '"${cmake_ld_flags} ${cmake_c_flags}"' -o test test.c
 '>"Makefile"
 echo '
 #include <stdio.h>
-- 
2.32.1 (Apple Git-133)

